chat/router.py

```diff
@@ -0,0 +1,125 @@
+from fastapi import APIRouter
+from pydantic import BaseModel
+
+from unsloth import FastLanguageModel
+import torch
+
+max_seq_length = 2048 # Choose any! We auto support RoPE Scaling internally!
+dtype = None # None for auto detection. Float16 for Tesla T4, V100, Bfloat16 for Ampere+
+load_in_4bit = True # Use 4bit quantization to reduce memory usage. Can be False.
+
+alpaca_prompt = """Below is an instruction that describes a task, along with an input that provides additional context. Write a response that appropriately completes the request.
+
+### Instruction:
+{}
+
+### Input:
+{}
+
+### Response:
+{}"""
+
+class Question(BaseModel):
+    query: str
+
+router = APIRouter()
+
+@router.post("/generate_answer")
+def generate_answer(value: Question):
```

Comment:
This looks like an extended version of `generate_bot_response` in `original_main.py` below, if that's the case then you should delete that one as it was based on your draft PR to chat activity.

Also, does the prompt you've used here provide better responses than the one you used earlier?

Reply:
> This looks like an extended version of `generate_bot_response` in `original_main.py` below, if that's the case then you should delete that one as it was based on your draft PR to chat activity.
> 
> Also, does the prompt you've used here provide better responses than the one you used earlier?

For the first question: This is not an extended version of generate_bot_response from original_main.py. This version utilizes the Unsloth library within Chat Activity and is significantly different from original_main.py. Additionally, we plan to remove the original_main.py file in the final modifications.

For the second question: Yes, the new prompt encourages the model to generate better responses.

Reply:
> For the first question: This is not an extended version of generate_bot_response from original_main.py. This version utilizes the Unsloth library within Chat Activity and is significantly different from original_main.py. Additionally, we plan to remove the original_main.py file in the final modifications.

Yes it's significantly different from `original_main.py`, but it's achieving the same goal as that so generate_bot_response from original_main.py needs to be deleted to avoid duplicity.

Reply:
> > For the first question: This is not an extended version of generate_bot_response from original_main.py. This version utilizes the Unsloth library within Chat Activity and is significantly different from original_main.py. Additionally, we plan to remove the original_main.py file in the final modifications.
> 
> Yes it's significantly different from `original_main.py`, but it's achieving the same goal as that so generate_bot_response from original_main.py needs to be deleted to avoid duplicity.

Oh, I agree! I will delete that!

Reply:
Thank you! Can you also start working on Kshitij's part? I think you can work with what he has so far.

Reply:
Ok, I will complete it after my final exam.